app/controllers/api/v2/compute_resources_controller.rb

```diff
@@ -78,8 +78,10 @@ def create
           return
         end
         datacenter = change_datacenter_to_uuid(compute_resource_params[:datacenter])
-        @compute_resource.datacenter = datacenter if datacenter.present?
-
+        if compute_resource_params["provider"].downcase == "ovirt"
+          @compute_resource = ComputeResource.new_provider(compute_resource_params.merge!(datacenter: datacenter))
```

Comment:
Maybe add a comment about why this is needed - since we need to create a new client with the datacenter uuid

Reply:
@tbrisker done